### ISSUE TITLE
shuffleSet fails with control

### ISSUE THREAD

**Issue by LindsayVass** [gavinsimpson/permute#9]
Thanks for the quick fix to the first issue. I'm now running into an error when I try to use shuffleSet on the same control:

```{r}
control <- how(within = Within(type = "free"),
                plots = Plots(strata = rep(gl(2,7),2), type = "free"),
                blocks = gl(2, 14))
permSet <- shuffleSet(28, 100, control = control)
```
```
Error in doShuffleSet(spln[[i]], nset = nset, control) : 
  number of items to replace is not a multiple of replacement length
```

**Comment by gavinsimpson** [comment]
Ok, thanks. Let me take a look; I suspect it has something to do with the plots being a single factor of two levels ...

**Comment by jarioksa** [comment]
Shouldn't plots be nested within blocks -- not crossed?

**Comment by gavinsimpson** [comment]
@jarioksa yes, but in one sense, the plot and block definitions here are:

```r
> rep(gl(2,7),2)
 [1] 1 1 1 1 1 1 1 2 2 2 2 2 2 2 1 1 1 1 1 1 1 2 2 2 2 2 2 2
Levels: 1 2
> gl(2,14)
 [1] 1 1 1 1 1 1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2 2 2 2 2
Levels: 1 2
```

If you consider `plots` to be an indicator, then it does index the two plots in each block. Ideally you'd do the plots as

```r
> gl(4,7)
 [1] 1 1 1 1 1 1 1 2 2 2 2 2 2 2 3 3 3 3 3 3 3 4 4 4 4 4 4 4
Levels: 1 2 3 4
```

but I know at least `shuffle` would complain with that because I forgot to drop the empty levels when I split plots up by blocks. Essentially for the first block we have plot

```r
> gl(4,7)[1:14]
 [1] 1 1 1 1 1 1 1 2 2 2 2 2 2 2
Levels: 1 2 3 4
```

and some bits of (IIRC) `shuffleStrata()` really don't like those missing levels.

**Comment by gavinsimpson** [comment]
This was a silly little bug; `shuffleStrata` was permuting *all* the plot strata, not the plot strata for the current block. Fixed in 98e179cf542ec9ea1d963edef377686906b61886

Thanks for the report @LindsayVass
